skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,281 @@
+import warnings
+from collections.abc import Sequence
+from numbers import Integral
+from typing import Any, Optional, Union
+
+from rdkit.Chem import AllChem, Mol
+from rdkit.SimDivFilters.rdSimDivPickers import MaxMinPicker
+from sklearn.utils._param_validation import Interval, RealNotInt, validate_params
+
+from skfp.model_selection.splitters.utils import (
+    create_dice_distance_function_from_fingerprints,
+    ensure_nonempty_subset,
+    get_data_from_indices,
+    split_additional_data,
+    validate_train_test_split_sizes,
+    validate_train_valid_test_split_sizes,
+)
+from skfp.utils.validators import ensure_mols
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "return_indices": ["boolean"],
+    },
+    prefer_skip_nested_validation=True,
+)
+def maxmin_train_test_split(
+    data: Sequence[Union[str, Mol]],
+    *additional_data: Sequence,
+    fingerprints=None,
+    train_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    return_indices: bool = False,
+    random_state: int = 0,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]], Sequence[Union[str, Mol]], Sequence[Sequence[Any]]
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int]],
+]:
+    """
+    Split using maxmin algorithm of picking a subset of item from a pool.
+    Starting from random item of initial set, next is picked item with maximum
+    value for its minimum distance to molecules in the picked set (hence the MaxMin name),
+    calculating and recording the distances as required. This molecule is the most distant
+    one to those already picked so is transferred to the picked set
+
```

Comment:
Add a very short single line at the beginning with minimal description. It is rendered in tables in the docs. See fingerprints or filters for examples.

Description should follow the general style of scaffold / Butina (see PR) docs, see them for examples. Dot is missing from the last sentence.